### PR TITLE
feat(odyssey-react): update Link to use scss module

### PR DESCRIPTION
**Pull request by arnoldsandoval-okta**
[OKTA-416111](https://oktainc.atlassian.net/browse/OKTA-416111)

### CODE REVIEW

File: packages/odyssey-react/.storybook/main.js [hunk 1]
```diff
@@ -47,8 +47,11 @@ module.exports = {
                 @import '@okta/odyssey/src/scss/base/reset';
                 @import '@okta/odyssey/src/scss/base/accessibility';
                 @import '@okta/odyssey/src/scss/base/iconography';
-                @import '@okta/odyssey/src/scss/base/typography';
-
+                @import '@okta/odyssey/src/scss/base/typography-global';
+                @import '@okta/odyssey/src/scss/base/typography-text';
+                @import '@okta/odyssey/src/scss/base/typography-list';
+                @import '@okta/odyssey/src/scss/base/typography-header';
```
Comment (arnoldsandoval-okta) [inline packages/odyssey-react/.storybook/main.js:53]: These were the imports previously found in `@okta/odyssey/src/scss/base/typography`, it also included a `typography-list` partial which is no longer necessary as this PR includes the module for it.

File: packages/odyssey-react/src/components/Link/Link.module.scss [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    color: cv('blue', '500');
+    text-decoration: underline;
+  }
+
+  &:visited {
+    color: cv('blue', '500');
+  }
+
+  /* stylelint-disable selector-no-qualifying-type */
+  &[target='_blank']:not([href^='mailto:'])::after {
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Link/Link.module.scss:27]: can we revisit this code here? I would love to see a separate class for this style, `linkExternal`. And have the application of this class entirely controlled through a prop for `Link` `external`

File: packages/odyssey-react/src/components/Link/Link.module.scss [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    inset-block-start: -0.1em;
+    width: 1em;
+    height: 1em;
+    margin-inline-start: 0.2em;
+    background-image: get-icon('external', $color-primary-base);
+    background-repeat: no-repeat;
+    vertical-align: middle;
+  }
+
+  /* stylelint-enable */
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Link/Link.module.scss:40]: as an aside: if you want to disable a naughty selector as we're doing here you can use

```
/* stylelint-disable-next-line <RULE NAME HERE> */
```

File: packages/odyssey-react/src/components/Link/Link.module.scss [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+.linkHasIcon {
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Link/Link.module.scss:43]: consider `LinkWithIcon` ?

File: packages/odyssey-react/.storybook/main.js (hunk 1 above)
Comment (jeffbelser-okta) [reply packages/odyssey-react/.storybook/main.js:53]: Thanks for the comment - eventually we'll have to work toward removing everything here but we are making slow and steady progress!

**Review by jeffbelser-okta — APPROVED** [review]
See my note about the external link CSS, I would love to fix that here

File: packages/odyssey-react/src/components/Link/Link.module.scss (hunk 2 above)
Comment (arnoldsandoval-okta) [reply packages/odyssey-react/src/components/Link/Link.module.scss:27]: Fixed, however, it is using the `target` prop to apply the `linkExternal` styling, if we want to change this I would suggest we do it as part of the "react next steps" epic.

File: packages/odyssey-react/src/components/Link/Link.module.scss (hunk 3 above)
Comment (arnoldsandoval-okta) [reply packages/odyssey-react/src/components/Link/Link.module.scss:40]: Good tip, this was removed as it was no longer necessary!

File: packages/odyssey-react/src/components/Link/Link.module.scss (hunk 4 above)
Comment (arnoldsandoval-okta) [reply packages/odyssey-react/src/components/Link/Link.module.scss:43]: updated

File: packages/odyssey-react/src/components/Link/index.tsx [hunk 5]
```diff
@@ -53,13 +53,19 @@ const Link: FunctionComponent<Props> = (props) => {
 
   const omitProps = useOmit(rest);
 
+  const componentClass = (() => {
+    const classes = [styles.link]
+    if (target === '_blank') classes.push(styles.linkExternal);
+    return classes.join(' ');
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Link/index.tsx:59]: why do this by hand rather than using our `useCx` utility?

Comment (jeffbelser-okta) [reply packages/odyssey-react/src/components/Link/index.tsx:59]: cc @arnoldsandoval-okta ☝️

**Pull request merged**